### ISSUE TITLE
No way to specify composite id

### ISSUE THREAD

**Issue by angstream555** [spring-projects/spring-data-r2dbc#158]
No way to specify composite id for usage ReactiveCrudRepository

**Comment by mp911de** [comment]
We depend on upstream feature availability in Spring Data Relational for composite Id's, see [DATAJDBC-352](https://jira.spring.io/browse/DATAJDBC-352).

**Comment by peterjurkovic** [comment]
any idea when it will be done?

**Comment by mp911de** [comment]
No, we don’t have this feature scheduled yet as we’re busy with other tasks. A pre-requisite is the composite I’d support in Spring Data Relational as the R2DBC module depends on a shared library.

**Comment by pwittchen** [comment]
Can you suggest any workaround for this until it's done?

**Comment by porfirioribeiro** [comment]
@pwittchen do you mind to explain how did you did it with Tuple2?

**Comment by pwittchen** [comment]
@peterjurkovic 

Something like this:

```java
@Repository
public interface UserRepository extends R2dbcRepository<User, Tuple2<BigDecimal, String> {
  //...
}
```

I know, this is kind of workaround and won't work with Repository methods provided out of the box like `findById(Id)` etc. and you need to write custom methods with custom queries anyway, but I keep it for code documentation purposes to avoid providing any (non-composite) key, which is not real database key to satisfy interface signature.

**Comment by RodRich1991** [comment]
this is still forgotten =/

**Comment by palaknagda** [comment]
@pwittchen Can you please give a reference to your code?

**Comment by pwittchen** [comment]
@palaknagda Code related to my comment is proprietary and is not open-source. You can see a general idea in my code snippet in the earlier comment. Nevertheless, it does not solve this problem and query inside `@Query` annotation should work regardless of this change.

**Comment by tiagorg-cit** [comment]
Any updates???

**Comment by mp911de** [comment]
Still depends on spring-projects/spring-data-relational#574. A well-crafted pull request along with a decent design can always speed things up.

**Comment by paulushcgcj** [comment]
More than 3 years later, no updates so far?

**Comment by ivandedkov** [comment]
Any updates? The related task above has been done
